src/Angular.js

```diff
@@ -767,6 +767,36 @@ function equals(o1, o2) {
   return false;
 }
 
+/**
+ * @ngdoc directive
+ * @name ng.directive:ngJq
+ *
+ * @element html
+ * @param {string} (optional) the name of the library available under `window` to be used for angular.element
+ *
+ * @description
+ *
+ * Use this directive to force the angular.element library.  This should be used to force
+ * either jqLite by leaving ng-jq blank or setting the name of the jquery variable under window
+ * (eg. jQuery).
+ *
+ * This directive is global for the whole of the angular library, hence the directive can only
+ * be added to the top-most HTML element.
+ *
+ */
+var jq = function () {
+  if (isDefined(jq.name_)) return jq.name_;
+
+  var el = document.querySelector('[ng-jq]') || document.querySelector('[data-ng-jq]');
```

Comment:
Crap.  That's weird.  Something must of messed up somewhere.

Let me add some of your suggestions and I'll create a new pull request based on the right parent.

Cheers.